subworkflows/local/check_submission_dependencies.nf

```diff
@@ -6,35 +6,149 @@
+                    data_directory: it.data_directory
+                ]
+            }
+        }.set{analysis_channels}
+
+        //files : it.meta.status=='pass' ? workflow.stubRun ? [] : it.analysis.files.splitCsv(sep:'\t',header:true).fileName.collect { file("${path_to_files_directory[0]}/$it") } : [],
+        //Validate clinical and analysis 
+        VALIDATE_CLINICAL(
+             analysis_channels
+        )
+        ch_versions = ch_versions.mix(VALIDATE_CLINICAL.out.versions)
 
-    ch_versions = Channel.empty()
+        //Update channel status based on VALIDATE_CLINICAL outcome
+        VALIDATE_CLINICAL.out.status.map{
+            meta, analysis, clinical , files, status_file ->
+            def status_value = status_file.text.contains('status: "FAILED"') ? 'failed' : 'pass'
+            def updated_meta = meta.clone()
 
-    // TODO nf-core: substitute modules here for the modules of your subworkflow
-    Channel.value(1).subscribe{println "CHECK_SUBMISSION_DEPENDENCIES helloA"}
+            updated_meta.status = (status_value == 'pass' && meta.status == 'pass') ? 'pass' : 'failed'
+            [
+                meta : updated_meta,
+                analysis : analysis,
+                clinical : clinical,
+                files : files
+            ]
+        }.set{validated_analysis_channels}
 
-    clinical_upload = [[{},null]]
-    entity_mapping = [[],{}]
-    unsuccessful_dependency = [[],{}]
-    molecular_files_to_upload = [[],{}]
+        //Submit data to clinical.submission
+        CLINICAL_SUBMISSION(validated_analysis_channels)
+
+        //Update channel status based on CLINICAL_SUBMISSION outcome
+         CLINICAL_SUBMISSION.out.status.map{
+            meta, analysis, clinical , files, status_file ->
+            def status_value = status_file.text.contains('status: "FAILED"') ? 'failed' : 'pass'
+            def updated_meta = meta.clone()
+
+            updated_meta.status = (status_value == 'pass' && meta.status == 'pass') ? 'pass' : 'failed'
+            [
+                meta : updated_meta,
+                analysis : analysis,
+                clinical : clinical,
+                files : files
+            ]
+        }.set{submitted_analysis_channels}
 
     emit:
-    // TODO nf-core: edit emitted channels
-    clinical_upload  // channel: [ val(meta), [csv] ] multiple CSVs per entity
-    molecular_files_to_upload // channel [ val(meta) [files] ] per analysis
-    entity_mapping         // channel: [ val(meta), [ csv ] ] relational mapping
-    unsuccessful_dependency // [val(meta),[csv]]
-    versions = ch_versions                     // channel: [ versions.yml ]
+        versions = ch_versions
+        analysis_channels = analysis_channels//format checked
+        validated_analysis_channels = validated_analysis_channels //format checked and validated
+        submitted_analysis_channels = submitted_analysis_channels//format checked, validated and submitted to clinical service
+
```

Comment:
Can I get a channel for all status.yml files from the subworkflow `check_submission_dependencies`?
For example: https://github.com/Pan-Canadian-Genome-Library/molecular-data-submission-workflow/blob/integrate_test/subworkflows/local/data_validation/main.nf#L117